backend/composer/services/cs_ingestion/cs_ingestion_services.py

```diff
@@ -0,0 +1,234 @@
+        logging.warning(f'Skip statement {statement[LABEL]} due to sex {statement[SEX][0]} not found in composer db')
+        return True
+    else:
+        return False
+    
+def has_invalid_species(statement):
+    if has_prop(statement[SPECIES]):
+        for s in statement[SPECIES]:
+            try:
+                Specie.objects.get(ontology_uri=s)
+                return False
+            except:
+                logging.warning(f'Skip statement {statement[LABEL]} due to specie {s} not found in composer db')
+                return True 
+    return False
+    
+def get_valid_phenotypes(statement):
+    valid_phenotypes = []
+    if has_prop(statement[PHENOTYPE]):
+        for p in statement[PHENOTYPE]:
+            try:
+                phenotype = Phenotype.objects.get(ontology_uri=p)
+                valid_phenotypes.append(phenotype)
+            except:
+                logging.warning(f'Skip phenotype {p} at statement {statement[LABEL]} not found in composer db ')
+    return valid_phenotypes
+
+
+def pick_first_phenotype(phenotypes_list, statement):
+    if len(phenotypes_list) == 0:
+        return None
+    elif len(phenotypes_list) > 1:
+        logging.warning(f'Multiple phenotypes found, ignore the following {phenotypes_list[1:]} at statement {statement[LABEL]}')
+    return phenotypes_list[0]
+
+def validate_statements(statement_list):
+    valid_statements = []
+    for statement in statement_list:
+        if has_multiple_destination(statement):
+            continue
+        if has_invalid_entities(statement):
+            continue
+        if has_invalid_sex(statement):
+            continue
+        if has_invalid_species(statement):
+            continue
+
+        valid_statements.append(statement)
+
+    return valid_statements
+
+
+def create_artifact_sentence(statement):
+    text = f'{statement[LABEL]} created from neurondm on {NOW}'
+    has_sentence_reference = len(statement[SENTENCE_NUMBER]) > 0
+    sentence, created = Sentence.objects.get_or_create(
+            doi__iexact=statement[ID],
+            defaults={"title": text, "text": text,
+            "doi": statement[ID],
+            "external_ref": statement[SENTENCE_NUMBER][0] if has_sentence_reference else None,
+            "batch_name": f"neurondm-{NOW}" if has_sentence_reference else None
+            },
+        )
+    if created:
+        logging.info(f"Sentence for neuron {statement[LABEL]} created.")
+        sentence.save()
+    return sentence
+
+
+def has_prop(prop):
+    return True if len(prop) > 0 else False
+
+def get_value_or_none(model, prop):
+    if has_prop(prop):
+        if model == AnatomicalEntity:
+            return model.objects.filter(ontology_uri=prop[0])[0]
+        else:
+            return model.objects.get(ontology_uri=prop[0])
+    else:
+        return None
+
+
+
+def map_circuit_type_uri_with_model(uri):
+    circuit_type = {
```

Comment:
@Anto8819 I suggest to make this a CONSTANT and move it up in the document (just below the imports)